### PR TITLE
Convert bad expressions for error recovery

### PR DESCRIPTION
**Pull request by 333fred**
We were not handling unconverted expressions in some bad code scenarios, leading to issues where bad unconverted objects slipped through and broke further downstream scenarios. Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1806208.

### CODE REVIEW

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for review.

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for review

File: src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IObjectCreationExpression.cs [hunk 1]
```diff
@@ -15351,5 +15351,181 @@ static void M(IA a)
...
+            var source = """
+                var c = /*<bind>*/new C() { F1 = 1, $"{asdf}", true switch { _ => false }, new() }/*</bind>*/;
+
+                class C
+                {
+                    public int F1;
+                }
+                """;
+
+            var expectedDiagnostics = new[]{
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IObjectCreationExpression.cs:15367]: >new[]{ [](http://example.com/codeflow?start=38&length=6)

nit: formatting

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 5)

**Comment by 333fred** [comment]
@dotnet/roslyn for a second review.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 5)

File: src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IObjectCreationExpression.cs (hunk 1 above)
Comment (333fred) [reply src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IObjectCreationExpression.cs:15367]:
```suggestion
            var expectedDiagnostics = new[]
            {
```

**Pull request merged**